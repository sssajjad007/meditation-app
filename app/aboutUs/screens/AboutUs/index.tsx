import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React from "react";
import { View, Image, Linking } from "react-native";
import {
  Button,
  Container,
  THEME,
  Typography,
  version,
} from "../../../library";
import * as Analytics from "expo-firebase-analytics";
import { styles } from "./styles";
import Communications from "react-native-communications";
import { Scroller } from "../../../library/Scroller";
import { digitsEnToFa } from "@persian-tools/persian-tools";
export function AboutUs() {
  async function callEvent() {
    try {
      await Analytics.logEvent("  customer_service_call");
    } catch (error) {
      // console.log(error);
    }
  }
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  return (
    <Container headerTitle={"درباره ما"}>
      <Scroller>
        <View style={styles.iconContainer}>
          <Image
            source={require("../../assets/Logo.png")}
            style={styles.imageSize}
            resizeMode={"contain"}
          />
        </View>
        <View style={styles.bodyContainer}>
          <Typography
            mode={"mediumBody18"}
            style={{ color: THEME.COLORS.LABEL.SECONDARY, textAlign: "right" }}
          >
            {`دنج اپلیکیشنی برای سلامتی ذهن و بدن شماست. در شرایط زندگی امروزی پیدا کردن زمانی برای اینکه قدری تمرکز کنیم و کمتر در افکار غرق باشیم تبدیل شده به یه کار خیلی سخت. ما باور داریم خالی کردن ۱۰ دقیقه در روز برای مدیتیشن امکان‌پذیره. هدف ما در دنج اینه که این فرصت رو به بهترین شکل براتون ایجاد کنیم. در دنج از اپ‌های Calm، Waking Up و Headspace استفاده شده و ترجمه تمام مدیتیشن‌ها در تیم دنج انجام گرفته. برخی از جلسات هم توسط تیم دنج نوشته شده‌اند. با دنج لذت حضور رو تجربه کنید. `}
          </Typography>
        </View>
        <View style={styles.buttonContianer}>
          <Button
            mode={"text"}
            rippleColor={"lightGrey"}
            size={"small"}
            onPress={() => {
              navigation.push("TermsScreen");
            }}
          >
            <Typography
              mode={"boldBody16"}
              style={{ color: THEME.COLORS.PRIMARY.NORMAL }}
            >
              {"قوانین دنج"}
            </Typography>
          </Button>
          <Typography
            mode={"regularSub12"}
            style={{ color: THEME.COLORS.LABEL.TERTIARY, bottom: 12 }}
          >
            {`نسخه ${digitsEnToFa(version)}`}
          </Typography>
          <Button
            mode={"contained"}
            size={"big"}
            rippleColor={"lightGrey"}
            onPress={async () => {
              await callEvent();
              Linking.openURL("tel:02191006901");
              // Communications.phonecall("02191006901", false);
            }}
          >
            {"تماس با ما"}
          </Button>
        </View>
      </Scroller>
    </Container>
  );
}
