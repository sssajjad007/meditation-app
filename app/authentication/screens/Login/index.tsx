import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React from "react";
import { View, Image } from "react-native";
import { Button, Typography, THEME, Container } from "../../../library";
import { LogoImage } from "../../assets";
import { styles } from "./styles";

export function Login() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  return (
    <Container style={styles.containerStyles}>
      <View style={{ flex: 1 }}>
        <Image style={styles.imageSize} source={LogoImage} />
      </View>
      <View style={{ flex: 0.5, justifyContent: "center" }}>
        <Typography
          mode={"mediumHeadline22"}
          style={{
            color: THEME.COLORS.LABEL.PRIMARY,
          }}
        >
          {"به دنج خوش اومدید !"}
        </Typography>
      </View>
      <View style={{ flex: 1 }}>
        <Typography
          mode={"mediumBody16"}
          style={{ color: THEME.COLORS.LABEL.SECONDARY }}
        >
          {"با دنج میتونید مدیتیشن رو یاد بگیرید و تمرین کنید."}
        </Typography>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          mode={"contained"}
          rippleColor={"lightGrey"}
          size={"big"}
          onPress={() => {
            navigation.push("Auth");
          }}
        >
          {"ورود"}
        </Button>
      </View>
    </Container>
  );
}
