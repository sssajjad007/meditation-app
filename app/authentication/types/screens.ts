export type tStep = "phone" | "genCode" | "name" | "genCodeResend"|"enterGenCode";
