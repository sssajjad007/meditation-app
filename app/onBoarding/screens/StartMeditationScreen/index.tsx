import { useNavigation } from "@react-navigation/core";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React, { useEffect, useState, useContext } from "react";
import { View, Text, Image } from "react-native";
import { Container, Paragraph, THEME, Typography } from "../../../library";
import { login } from "../../../library/authEvent";
import * as Analytics from "expo-firebase-analytics";
// import TrackPlayer, {
//   Capability,
//   useProgress,
//   useTrackPlayerEvents,
//   Event,
// } from "react-native-track-player";
import { styles } from "./styles";
import { CloseLightIcon, PauseIcon } from "../../../library/Icon";
import Slider from "@react-native-community/slider";

import { widthPercentageToDP } from "react-native-responsive-screen";
import { MusicBackground } from "../../assets";
const tracks = [
  {
    id: 1,
    url: "https://denjapp.com/public/uploads/track/1640727501593.mp3",
    title: "first song",
    artist: "deadmau5",
    album: "while(1<2)",
    genre: "Progressive House, Electro House",
    date: "2014-05-20T07:00:00+00:00", // RFC 3339
    artwork: "https://cdn01.zoomit.ir/2021/10/great-wall.jpg?w=1300", // Load artwork from the network
  },
];
// TrackPlayer.updateOptions({
//   stopWithApp: false,
//   capabilities: [
//     Capability.Play,
//     Capability.Pause,
//     // Capability.Stop,
//   ],
//   compactCapabilities: [Capability.Play, Capability.Pause],
// });

export function StartMeditation() {
  //   async function playEvent() {
  //     try {
  //       await Analytics.logEvent("onboarding_played");
  //     } catch (error) {
  //       console.log(error);
  //     }
  //   }
  //   async function skipEvent() {
  //     try {
  //       await Analytics.logEvent("onboarding_skipped");
  //     } catch (error) {
  //       console.log(error);
  //     }
  //   }
  //   async function finishEvent() {
  //     try {
  //       await Analytics.logEvent("onboarding_finished");
  //     } catch (error) {
  //       console.log(error);
  //     }
  //   }
  //   const setUpTrackPlayer = async () => {
  //     try {
  //       await TrackPlayer.setupPlayer();
  //       await TrackPlayer.add(tracks);
  //       console.log("Tracks added");
  //     } catch (e) {
  //       console.log(e);
  //     }
  //   };
  //   const [playerState, setPlayerState] = useState<string>("");
  //   const events = [Event.PlaybackError, Event.RemotePlay, Event.RemotePause];
  //   useTrackPlayerEvents(events, (event) => {
  //     if (event.type === Event.PlaybackError) {
  //       console.warn("An error occured while playing the current track.");
  //     }
  //     if (event.type === Event.RemotePause) {
  //       setPlayerState("paused");
  //     }
  //     if (event.type === Event.RemotePlay) {
  //       setPlayerState("playing");
  //     }
  //   });
  //   useEffect(() => {
  //     setUpTrackPlayer();

  //     return () => TrackPlayer.destroy();
  //   }, []);

  //   const { duration, position } = useProgress();
  //   useEffect(() => {
  //     if (duration > 0) {
  //       const intDuration = Math.floor(duration);
  //       const intPosition = Math.floor(position);
  //       if (intDuration === intPosition) {
  //         OnEnd();
  //       }
  //     }
  //   }, [duration, position]);
  //   // const { setValue } = useContext(AuthContext);
  //   async function OnEnd() {
  //     await finishEvent;
  //     login();
  //   }
  //   function toTime(timeInSec: number) {
  //     const time = Math.floor(timeInSec);
  //     const minutes = Math.floor(time / 60);
  //     const second = time % 60;
  //     return `${minutes < 10 ? `0${minutes}` : `${minutes}`}:${
  //       second < 10 ? `0${second}` : `${second}`
  //     }`;
  //   }
  //   const isPlaying = playerState === "playing";
  //   async function onToggle() {
  //     try {
  //       if (isPlaying) {
  //         await TrackPlayer.pause();
  //         setPlayerState("paused");
  //       } else {
  //         await TrackPlayer.play();
  //         setPlayerState("playing");
  //       }
  //     } catch (error) {
  //       console.log(error);
  //     }
  //   }
  //   async function onSlidingComplete(value: number) {
  //     await TrackPlayer.seekTo(value);
  //   }
  //   // async function forward() {
  //   //   const newPosition = position + 15;
  //   //   if (newPosition > duration) {
  //   //     await TrackPlayer.seekTo(duration);
  //   //   } else {
  //   //     await TrackPlayer.seekTo(newPosition);
  //   //   }
  //   // }
  //   // async function backward() {
  //   //   const newPosition = position - 15;
  //   //   if (newPosition < 0) {
  //   //     await TrackPlayer.seekTo(0);
  //   //   } else {
  //   //     await TrackPlayer.seekTo(newPosition);
  //   //   }
  //   // }
  //   const navigation = useNavigation<NativeStackNavigationProp<any>>();
  //   const [play, setPlay] = useState<boolean>(false);
  return (
    <Container style={{ backgroundColor: THEME.COLORS.GREY.BACKGROUND }}>
      <View style={{ position: "absolute", top: 32, right: 24, zIndex: 10 }}>
        {/* <CloseLightIcon
          size={16}
          onPress={async () => {
            await skipEvent;
            login();
          }}
        /> */}
      </View>
      <View style={styles.titleContainer}>
        {/* {play ? (
          <View
            style={{
              top: 80,
              height: 380,
              alignItems: "center",
              justifyContent: "center",
            }}
          >
            <Image
              style={styles.image}
              resizeMode={"contain"}
              source={MusicBackground}
            />
            <View
              style={{
                position: "absolute",
                right: widthPercentageToDP(50) - 144 / 2 - 6,
              }}
            >
              {isPlaying ? (
                <PauseIcon size={144} name={"Pause"} onPress={onToggle} />
              ) : (
                <PauseIcon size={144} name={"Play"} onPress={onToggle} />
              )}
            </View>
          </View>
        ) : (
          <View style={{ top: "60%" }}>
            <Typography mode={"regularHeadline22"} style={[styles.textAlign]}>
              {"صدای گوشیتون رو روشن کنید و بیاید از اینجا شروع کنیم."}
            </Typography>
          </View>
        )} */}
      </View>
      <View style={styles.meditationPlayerContainer}>
        {true ? (
          <View style={styles.controller}>
            <View style={styles.duration}>
              {/* <Paragraph style={{ color: "white" }}>
                {toTime(position)}
              </Paragraph>
              <Paragraph style={{ color: "white" }}>
                {toTime(duration)}
              </Paragraph> */}
            </View>
            {/* <Slider
              style={styles.slider}
              value={position}
              onSlidingComplete={onSlidingComplete}
              minimumValue={0}
              maximumValue={duration}
              minimumTrackTintColor={THEME.COLORS.PRIMARY.NORMAL}
              maximumTrackTintColor={THEME.COLORS.GREY.ICON}
              thumbTintColor={THEME.COLORS.PRIMARY.LIGHT}
              step={1}
            /> */}
            {/* <View style={styles.actions}>
              <ShufleIcon onPress={backward} size={20} />
              <BackwardIcon onPress={backward} size={40} />
              <PauseIcon
                onPress={onToggle}
                name={isPlaying ? "Pause" : "Play"}
                size={80}
              />
              <ForwardIcon onPress={forward} size={40} />
              <RepeatIcon onPress={forward} size={20} />
            </View> */}
          </View>
        ) : (
          <View style={{ marginBottom: 100 }}>
            {/* <PauseIcon
              size={110}
              onPress={async () => {
                onToggle();
                await playEvent();
                setPlay(true);
              }}
            /> */}
          </View>
        )}
      </View>
    </Container>
  );
}
