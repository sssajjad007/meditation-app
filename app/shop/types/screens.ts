export interface IUseCafeBazaar {
  setSubscriptionDaysLeft: (daysLeft: number) => void;
  setUserPremium: (isPremium: boolean) => void;
}
