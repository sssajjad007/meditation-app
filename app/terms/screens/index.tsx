import { View, Text } from "react-native";
import React from "react";
import { styles } from "./styles";
import { Container, Paragraph, THEME, Typography } from "../../library";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { Scroller } from "../../library/Scroller";

export function TermsScreen() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  return (
    <Container
      headerTitle={"قوانین دنج"}
      headerTitleColor={"white"}
      backgroundColor={THEME.COLORS.LABEL.PRIMARY}
    >
      <Scroller
        contentContainerStyle={{ backgroundColor: THEME.COLORS.LABEL.PRIMARY }}
      >
        {/* <View style={styles.header}>
        <Title style={styles.toRight}>{"قوانین دنج"}</Title>
        <View style={styles.backIcon}>
          <ArrowRightIcon
            size={30}
            onPress={() => {
              navigation.goBack();
            }}
          />
        </View>
      </View> */}
        <View style={styles.termContainere}>
          <Typography
            mode={"regularBody14"}
            style={{ color: "white", textAlign: "right" }}
          >
            {`ماده ۱. این سیاست‌نامه حفظ حریم خصوصی برای محصول دنج ارائه شده توسط شرکت کاوشگران رادین فناور سلامت با نام تجاری کرفس است. دنج یک اپلیکیشن مدیتیشن است که با هدف آموزش و تسهیل در تمرین مدیتیشن در بین فارسی‌زبانان ایجاد شده است.`}
            {`
          
ماده ۲. قبل از استفاده از برنامه کاربران ملزم به ثبت‌نام می‌باشند. ثبت نام در دنج رایگان است. در قسمت ثبت‌نام اسم و شماره موبایل کاربر پرسیده می‌شود. اطلاعات دیگر از جمله سن و جنسیت از قسمت پروفایل در اپلیکیشن دنج قابل تکمیل شدن است. جمع‌آوری این اطلاعات به دنج اجازه می‌دهد تا دید دقیق‌تری نسبت به کاربران خود داشته باشد تا علاوه بر تعامل راحت‌تر، امکانات مناسب‌تری را برای‌ آن‌ها فراهم کند. دنج این اطلاعات را بدون اجازه‌ی کاربر با هیچ شخص ثالثی در میان نخواهد گذاشت.`}
            {`

ماده ۳. کاربر، مسئول حفظ و نگهداری سخت‌افزار کامپیوتر، تلفن‌ها و سایر ابزار ضروری برای دسترسی و استفاده از این سایت و تمام هزینه‌های مرتبط به آن بوده و دنج مسئول صدمات ناشی از ابزار کاربر نیست.
`}
            {`
ماده ۴. دنج اقدامات امنیتی لازم و مناسبِ سخت‌افزاری و نرم‌افزاری را برای جلوگیری از دسترسی‌های غیرمجاز و افشاء یا تخریب اطلاعات شخصی کاربر انجام می‌دهد. دنج اجازه دسترسی به اطلاعات شخصی کاربران را به کارکنان این شرکت می‌دهد اما این افراد با تعهدات محرمانه مواجه هستند و در صورت تخطی از این تعهدات با پیگرد قانونی مواجه خواهند شد. همچنین مطالعات آماری در مورد جامعه‌ی کاربران دنج بدون استفاده از هویت‌ شخصی آن‌ها و کاملاً ناشناس انجام می‌شود.`}
            {`

ماده ۵. دنج ممکن است با شرکت‌ها، وبسایت‌ها یا اپلیکیشن‌های دیگر همکاری کند اما هیچ‌گونه مسئولیتی در برابر سیاست‌های حفظ حریم شخصی و کاربری آنها ندارد. در صورتی که تمایل دارید از طریق اپلیکیشن دنج به سایت‌ها یا اپلیکیشن‌های دیگر وارد شوید، برای اطلاع از سیاست‌های حفظ حریم خصوصی این مجموعه‌ها به وب‌سایت آنها مراجعه کنید.`}
            {`

ماده ۶. تنها افراد بالای ۱۸ سال حق انجام معامله (خرید) را در اپلیکیشن دنج دارند.`}
            {`

ماده ۷. محصول خریدنی در دنج اشتراک است که با تهیه آن کاربر به تمام محتوا و امکانات اپ دسترسی پیدا می‌کند.
`}

            {`
ماده ۸. فرایند خرید در دنج به شرح زیر است:
انتخاب بسته‌ی اشتراکی مورد نظر
مشاهده‌ی پیش‌فاکتور
پرداخت از طریق درگاه اینترنتی
دریافت بسته‌ی اشتراکی`}
            {`

ماده ۹. اختلال در سرویس اینترنت یا هر دلیلی که باعث شود فرایند خرید به انجام نرسد باعث از بین رفتن حق کاربر نمی‌شود و در صورتی که مبلغی از حساب کاربر کسر شده باشد از طریق بانک به وی بازپرداخته خواهد شد.`}
            {`

ماده ۱۰. اطلاعات شخصی کاربران شامل نام، شماره‌ی تلفن و رمز عبور آنها نزد دنج محفوظ است و آن را به شخص یا شرکت دیگری واگذار نخواهد کرد.`}
            {`

ماده ۱۱. دنج بر مبنای قوانین جرايم رایانه‌ای متعهد می‌شود با استفاده از تمامی امکانات و منابع خود در جهت حفظ و حراست از اطلاعات شخصی کاربران اقدام نماید. بدیهی است که استفاده‌ی مداوم از برنامه به منزله‌ی تایید شرایط سیاست‌نامه‌ی حفظ حریم خصوصی دنج است.`}
            {`

ماده ۱۲. دنج تعهدی در موثر بودن مدیتیشن برای کاربران ندارد. کیفیت اجرای تمرین، پیوسته بودن تمرین و ویژگی‌های شخصی می‌تواند نتایج را تحت تاثیر قرار دهد.`}
            {`
ماده ۱۳. در صورت داشتن هرگونه ابهام، سوالات خود را به یکی از روش‌های زیر با ما در میان بگذارید:

پشتیبانی: 02191006901

ایمیل دنج : app.denj@gmail.com

آدرس: تهران، میدان ونک، خیابان برزیل، خیابان شمس لاهیجانی، بن‌بست ۱۲، پلاک `}
          </Typography>
        </View>
      </Scroller>
    </Container>
  );
}
