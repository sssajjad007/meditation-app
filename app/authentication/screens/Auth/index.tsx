import React, { useEffect, useState } from "react";
import { View, Image, KeyboardAvoidingView, Platform } from "react-native";
import { Container, THEME, Typography } from "../../../library";
import { login } from "../../../library/authEvent";
import { LogoImage } from "../../assets";
import { AuthButton } from "../../components/AuthButton";
import { AuthInput } from "../../components/AuthInput";
import { styles } from "./styles";
import { register, login as loginUsecase } from "../../usecases";
import { AuthTitle } from "../../components/AuthTitle";
import { tStep } from "../../types";
import * as Analytics from "expo-firebase-analytics";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { SafeAreaView } from "react-native-safe-area-context";

import { useOtpVerify } from "./useOtpVerify";

export function Authentication() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const [phone, setPhone] = useState<string>("");
  const [genCode, setGenCode] = useState<string>("");
  const [step, setStep] = useState<tStep>("phone");
  const [error, setError] = useState<string>("");
  const [otpError, setOtpError] = useState<string>("");
  const [show, setShow] = useState<Boolean>(true);
  const [name, setName] = useState<string>("");
  async function nameEvent() {
    try {
      await Analytics.logEvent("registeration_page2_view");
    } catch (error) {
      // console.log(error);
    }
  }
  async function phoneNumberEvent() {
    try {
      await Analytics.logEvent("registeration_page1_view");
    } catch (error) {
      // console.log(error);
    }
  }
  async function onRegisterPress() {
    const registered = await register(phone);
    if (registered) {
      setStep("genCode");
    }
  }
  async function onLoginPress() {
    const loggedin: string = await loginUsecase(phone, genCode);
    if (loggedin === "error") {
      setOtpError("کد تایید اشتباهه!");
      return;
    }
    if (loggedin === "COMPLETE_USER") {
      login();
      return;
    }
    if (loggedin === "NEW_USER") {
      setStep("name");
      await nameEvent();
      return;
    }
  }
  useEffect(() => {
    phoneNumberEvent();
  }, []);
  useEffect(() => {
    if (genCode.length === 4) {
      onLoginPress();
    }
  }, [genCode]);
  useOtpVerify(setGenCode);
  return (
    <SafeAreaView style={{ flex: 1, backgroundColor: THEME.COLORS.BACKGROUND }}>
      <View style={styles.logoContainer}>
        <Image
          resizeMode={"stretch"}
          style={styles.imageSize}
          source={LogoImage}
        />
      </View>
      <View style={styles.textContainer}>
        <AuthTitle step={step} />
      </View>
      <View style={styles.inputContainer}>
        <AuthInput
          error={error}
          otpError={otpError}
          setError={setError}
          setStep={setStep}
          step={step}
          phone={phone}
          genCode={genCode}
          setGenCode={setGenCode}
          setPhone={setPhone}
          setShow={setShow}
          name={name}
          setName={setName}
        />
      </View>
      <KeyboardAvoidingView
        style={styles.buttonContainer}
        behavior={Platform.OS === "android" ? "height" : "padding"}
      >
        <AuthButton
          error={error}
          phone={phone}
          setStep={setStep}
          onLoginPress={onLoginPress}
          onRegisterPress={onRegisterPress}
          step={step}
          name={name}
          setName={setName}
        />
        {step === "phone" ? (
          <Typography
            mode={"regularSub12"}
            style={{ paddingTop: 10, color: THEME.COLORS.LABEL.SECONDARY }}
          >
            با ثبت نام و ورود با
            <Typography
              onPress={() => {
                navigation.push("TermsScreen");
              }}
              mode={"boldSub12"}
              style={{ color: THEME.COLORS.PRIMARY.NORMAL }}
            >
              {" "}
              قوانین دنج{" "}
            </Typography>
            موافقت می کنید.
          </Typography>
        ) : undefined}
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}
