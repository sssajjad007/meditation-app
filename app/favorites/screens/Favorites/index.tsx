import { useIsFocused, useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React, { useEffect, useState } from "react";
import { View, Text, ActivityIndicator } from "react-native";
import { Container, THEME, Typography } from "../../../library";
import { LikeIcon } from "../../../library/Icon";
import { Scroller } from "../../../library/Scroller";
import { FavCard } from "../../components/FavCard";
import { IFavTracksList } from "../../types/usecases";
import { deleteFavTracks } from "../../usecases";
import { retrieveFavTracks } from "../../usecases/getFavTracks";
import { styles, THEME_COLORS_PRIMARY_NORMAL } from "./styles";

export function Favorites() {
  const isFocused = useIsFocused();
  const [empty, setEmpty] = useState<boolean>(true);
  const [tracks, setTracks] = useState<IFavTracksList[]>([]);
  const [change, setChange] = useState<boolean>(false);
  const [loading, setLoading] = useState<boolean>(true);
  const navigation = useNavigation<NativeStackNavigationProp<any>>();

  async function retrieveFavTrack() {
    const favTracksList = await retrieveFavTracks();
    setTracks(favTracksList);
    setLoading(false);
    if (favTracksList.length === 0) {
      setEmpty(true);
    } else {
      setEmpty(false);
    }
  }
  async function unFavPress(objectId: string) {
    setChange(true);
    await deleteFavTracks(objectId);
    setChange(false);
  }
  useEffect(() => {
    retrieveFavTrack();
  }, [change, isFocused]);
  function renderFavTrack() {
    const result: JSX.Element[] = [];
    for (let index = 0; index < tracks.length; index++) {
      const { objectId, title, type, url } = tracks[index];
      result.push(
        <FavCard
          objectId={objectId}
          url={url}
          key={objectId}
          title={title}
          description={type}
          onPress={onCardPress}
          unLikePress={unFavPress}
        />
      );
    }
    return result;
  }
  async function onCardPress(objectId: string, title: string, url: string) {
    navigation.push("Player", { id: objectId, title: title, url: url });
  }
  function render() {
    if (empty && !loading) {
      return (
        <View style={styles.empty}>
          <LikeIcon size={134} />
          <Typography
            mode={"regularSub12"}
            style={{ color: THEME.COLORS.LABEL.SECONDARY }}
          >
            {"فعلا جلسه ای رو لایک نکردید!"}
          </Typography>
        </View>
      );
    }
    if (!empty && !loading) {
      return <Scroller>{renderFavTrack()}</Scroller>;
    }
    if (loading) {
      return (
        <View style={styles.loading}>
          <ActivityIndicator size="large" color={THEME_COLORS_PRIMARY_NORMAL} />
        </View>
      );
    }
  }
  return <Container headerTitle={"جلسات مورد علاقه"}>{render()}</Container>;
}
