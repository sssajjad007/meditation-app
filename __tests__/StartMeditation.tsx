import React from "react";
import renderer from "react-test-renderer";
import { StartMeditation } from "../app/onBoarding/screens/StartMeditationScreen";

test("renders correctly", () => {
  const tree = renderer.create(<StartMeditation />).toJSON();
  expect(tree).toMatchSnapshot();
});
