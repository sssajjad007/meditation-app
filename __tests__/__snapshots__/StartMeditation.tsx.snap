// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "top",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F9FAFC",
          "flex": 1,
          "paddingBottom": 40,
          "paddingTop": 32,
        },
        undefined,
        Object {
          "backgroundColor": "#061442",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "position": "absolute",
          "right": 24,
          "top": 32,
          "zIndex": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 4,
          "justifyContent": "center",
          "paddingHorizontal": 12,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 5,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column-reverse",
            "height": 130,
            "justifyContent": "space-evenly",
            "width": 705,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": "8%",
              "width": 645,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
